### PR TITLE
Add crun based jobs

### PR DESCRIPTION
**Pull request by harche**
This PR add [crun](https://github.com/containers/crun) based jobs. 

Signed-off-by: Harshal Patil <harpatil@redhat.com>

### CODE REVIEW

**Comment by harche** [comment]
Hello @giuseppe. Right now RHCOS ships an older version of crun (0.20.1), but we want to make sure that we are testing the latest crun with these jobs. 

@miabbott is working on getting latest crun (1.2.x) packages into RHCOS. Once they are in, let's observe how these tests perform.

**Comment by giuseppe** [comment]
> Hello @giuseppe. Right now RHCOS ships an older version of crun (0.20.1), but we want to make sure that we are testing the latest crun with these jobs.

Thanks for the update!  I am just wondering if there is any small crun vs runc difference that prevents some pods to run, and it will be faster to get a working version if I could fix these earlier.  There were few in the past that we discovered through the CRI-O e2e tests.

**Comment by miabbott** [comment]
@harche FYI RHCOS 410.84.202112011821-0 and all future RHCOS 4.10 builds should include `crun`

**Comment by giuseppe** [comment]
the CI failures should be fixed in master now

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by harche** [comment]
@staebler @deads2k need approval for adding crun jobs.

File: ci-operator/step-registry/openshift/manifests/crun/openshift-manifests-crun-commands.sh
Comment (wking) [inline ci-operator/step-registry/openshift/manifests/crun/openshift-manifests-crun-commands.sh:32]: nit: might be slightly easier to maintain going foward if you keep the `manifest_mc-master-crun.yml` as you have it, and then:

```sh
sed 's/master/worker/g' "${SHARED_DIR}/manifest_mc-master-crun.yml" > "${SHARED_DIR}/manifest_mc-worker-crun.yml"
```

Or some such to write the control plane manifest.

File: ci-operator/step-registry/openshift/manifests/crun/openshift-manifests-crun-commands.sh
Comment (wking) [inline ci-operator/step-registry/openshift/manifests/crun/openshift-manifests-crun-commands.sh:21]: nit: probably easier to maintain if you write this to a `/tmp` file and then use `base64` to encode it.  Something like:

```sh
cat <<EOF >/tmp/50-crun
[crio.runtime]
default_runtime = "crun"
[crio.runtime.runtimes.crun]
runtime_root = "/run/crun"
EOF

cat <<EOF > "${SHARED_DIR}/manifest_mc-master-crun.yml"
apiVersion: machineconfiguration.openshift.io/v1
...
          source: data:text/plain;charset=utf-8;base64,$(base64 -w0 </tmp/50-crun)
...
EOF
```

**Review by wking — APPROVED** [review]
/lgtm
/approve

And in case you want to pick up my inline suggestions, I'll drop a hold:

/hold

Feel free to just drop the hold, if the suggestions don't look useful

**Comment by harche** [comment]
Thanks @wking for the suggestions. I have updated the PR.

**Pull request merged**
